minio/README.md

```diff
@@ -0,0 +1,83 @@
+## About
+ MariaDB Foundation configuration of a Single-Node, Single-Drive MinIO deployment for running server S3 integration tests in BuildBot.
+
+Available at: [minio.mariadb.org](https://minio.mariadb.org)
+
+## Usage
+
+**Docker-compose.yaml** consists of two services:
+
+* **NGINX**, responsible for the SSL termination and the routing to the appropriate locations for the console and the API, see **minio.conf**
+
+* A single node/drive **MinIO** instance configured to expose the **console** and **API** on HTTP ports 8080/8081. The container is bind mounted to a data directory for persistent state (buckets, user configuration, and so on).
+
+**To run it locally**, one can only spin-up the MinIO service and access it on localhost, without any other prerequisites. Without providing the ENV variables ```MINIO_ROOT_{USER,PASSWORD}```, default credentials are used. Please consult the official documentation for more details.
+
+### Service stop/start
+```
+docker-compose down
+docker-compose --env-file .env -d
+```
+
+## Prerequisites
+
+### Layout
+```
+minio/
+└── docker-compose
+    ├── certbot
+    │   ├── ssl
+    │   │   ├── dhparam
+    │   │   ├── fullchain.pem
+    │   │   └── privkey.pem
+    │   └── www
+    ├── docker-compose.yaml
+    ├── logs
+    │   └── nginx
+    │       ├── access.log
+    │       ├── error.log
+    │       ├── minio.mariadb.org.access.log
+    │       └── minio.mariadb.org.error.log
+    └── nginx
+        ├── conf.d
+        │   └── minio.conf
+        └── nginx.conf
+```
+
+
+### TLS configuration
+LE certificates are provided by **Certbot** with the **Webroot** plugin. This assumes NGINX is configured to handle ACME challenges on port 80.
+
+```
+certbot certonly --webroot \ 
+--webroot-path #path_to_nginx_webroot# \
+ -d minio.mariadb.org \ 
+ -d minio.dev.mariadb.org
+ ```
+
+Example force renewal:
+```
+certbot renew --force-renewal --post-hook "bash /etc/letsencrypt/post_hook.sh"
+```
+
+The **post-hook** is configured to:  
+ -> transfer the renewed certificates to nginx path   
+ -> restart nginx container
+
+ Make sure **dhparam** is available for NGINX.  
+ ```
+ curl https://ssl-config.mozilla.org/ffdhe2048.txt > /path/to/dhparam
+ ```
+
+### Cron (renewal)
+
+To renew the certificates, define a certbot file under **cron.d**
+```
+0 */12 * * * root perl -e 'sleep int(rand(43200))' && certbot -q renew --post-hook "bash /etc/letsencrypt/post_hook.sh"
+```
```

Comment:
This should be run on the host (no cron on containers), and the line should be adapted then.

Reply:
The cron is defined on the host, I need to specify this.

Reply:
ok maybe it's worth handling this with ansible, I think that we might end up having similar deployments and it will probably help us (and be self documented then).

Reply:
Yep, better in Ansible.